### PR TITLE
[SPARK-21897][PYTHON][R] Add unionByName API to DataFrame in Python and R

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes to add a wrapper for `unionByName` API to R and Python as well.

**Python**

```python
df1 = spark.createDataFrame([[1, 2, 3]], ["col0", "col1", "col2"])
df2 = spark.createDataFrame([[4, 5, 6]], ["col1", "col2", "col0"])
df1.unionByName(df2).show()
```

```
+----+----+----+
|col0|col1|col3|
+----+----+----+
|   1|   2|   3|
|   6|   4|   5|
+----+----+----+
```

**R**

```R
df1 <- select(createDataFrame(mtcars), "carb", "am", "gear")
df2 <- select(createDataFrame(mtcars), "am", "gear", "carb")
head(unionByName(limit(df1, 2), limit(df2, 2)))
```

```
  carb am gear
1    4  1    4
2    4  1    4
3    4  1    4
4    4  1    4
```

## How was this patch tested?

Doctests for Python and unit test added in `test_sparkSQL.R` for R.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @felixcheung, could you take a look please?

**Comment by viirya** [comment]
Python side change LGTM

**Comment by felixcheung** [comment]
do you mean for your example in the PR description something different?
it says
```
head(union(limit(df1, 2), limit(df2, 2)))
```

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -2714,6 +2714,40 @@ setMethod("unionAll",
             union(x, y)
           })
 
+#' Return a new SparkDataFrame containing the union of rows
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: I think I'd suggest a slight different in the title - this one is the same words as `union`

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -2714,6 +2714,40 @@ setMethod("unionAll",
...
+#'
+#' Return a new SparkDataFrame containing the union of rows in this SparkDataFrame
+#' and another SparkDataFrame. This is different from both \code{UNION ALL} and
+#' \code{UNION DISTINCT} in SQL as column positions are not taken into account.
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: I'd list `union()` here too

**Review by felixcheung — APPROVED** [review]
LGTM, minor comments

**Comment by HyukjinKwon** [comment]
Oh, yea. I made a mistake during copying and pasting the example in the description.

File: R/pkg/R/DataFrame.R (hunk 1 above)
Comment (HyukjinKwon) [reply R/pkg/R/DataFrame.R]: Sure, I just addressed both comments.

**Review by felixcheung — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thank you @viirya and @felixcheung.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
I am hitting a network issue. Will update JIRA tomorrow.

**Pull request closed without merging**